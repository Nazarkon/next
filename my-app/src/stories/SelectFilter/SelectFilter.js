'use client'
import React from 'react';
import PropTypes from 'prop-types';

import './SelectFilter.scss';

const SelectFilter = ({ filterOptions, currentSelectedType, onChange }) => {
  return (
    <label className="select-label">
      <span className="select-label-text">SORT BY:</span>
      <select
        className="select-component"
        value={currentSelectedType}
        onChange={(e) => onChange(e.target.value)}
      >
        {filterOptions.map((option) => (
          <option key={option.id} className="select-option" value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </label>
  );
};

SelectFilter.propTypes = {
  filterOptions: PropTypes.array.isRequired,
  currentSelectedType: PropTypes.string.isRequired,
  onChange: PropTypes.func
};

export default SelectFilter;
