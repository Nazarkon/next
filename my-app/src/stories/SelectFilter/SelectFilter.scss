.select-label-text{
    font-weight: 400;
    font-size: 16px;
    color: #E4E4E4;
}
.select-component{
    margin-left: 30px;
    border: none;
    background-color: transparent;
    font-size: 16px;
    color: #FFFFFF
}