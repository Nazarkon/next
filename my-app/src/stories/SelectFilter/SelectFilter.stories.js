import React, { useState } from 'react';

import SelectFilter from './SelectFilter';

export default {
  title: 'Components/SelectFilter',
  component: SelectFilter,
  argTypes: {
    filterOptions: {
      control: 'array',
      description: 'Array of filter options',
      defaultValue: [
        {
          id: 1,
          label: 'RELEASE DATE',
          value: 'date'
        },
        {
          id: 2,
          label: 'TITLE',
          value: 'title'
        }
      ]
    },
    currentSelectedType: {
      control: 'text',
      description: 'Current selected sort type',
      defaultValue: 'title'
    }
  }
};

const Template = (args) => {
  const [currentItem, setCurrentItem] = useState(args.currentSelectedType);

  const handleChange = (value) => {
    setCurrentItem(value);
  };

  return <SelectFilter {...args} currentSelectedType={currentItem} onChange={handleChange} />;
};

export const Default = Template.bind({});
