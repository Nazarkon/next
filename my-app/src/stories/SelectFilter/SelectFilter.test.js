import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import SelectFilter from './SelectFilter';
import { userEvent } from '@storybook/testing-library';

describe('SelectFilter component', () => {
  const mockFilterOptions = [
    {
      id: '1',
      label: 'RELEASE DATE',
      value: 'date'
    },
    {
      id: '2',
      label: 'TITLE',
      value: 'title'
    }
  ];
  it('Matches snapshot', () => {
    const { container } = render(
      <SelectFilter
        filterOptions={mockFilterOptions}
        currentSelectedType={mockFilterOptions[0].value}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('Check that label and options renders correctly', () => {
    const { getByLabelText, getAllByRole } = render(
      <SelectFilter
        filterOptions={mockFilterOptions}
        currentSelectedType={mockFilterOptions[0].value}
      />
    );

    const label = getByLabelText('SORT BY:');
    expect(label).toBeInTheDocument();

    const options = getAllByRole('option');
    expect(options).toHaveLength(2);
    expect(options[0]).toHaveTextContent('RELEASE DATE');
    expect(options[1]).toHaveTextContent('TITLE');
  });

  it('Check that onChange handler calls when option is selected', () => {
    const handleChange = jest.fn();
    const { getByRole } = render(
      <SelectFilter
        filterOptions={mockFilterOptions}
        currentSelectedType={mockFilterOptions[0].value}
        onChange={handleChange}
      />
    );

    const select = getByRole('combobox');
    userEvent.selectOptions(select, 'date');

    expect(handleChange).toHaveBeenCalledWith('date');
  });
});
