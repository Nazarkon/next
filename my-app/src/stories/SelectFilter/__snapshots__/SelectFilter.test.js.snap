// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter component Matches snapshot 1`] = `
<div>
  <label
    class="select-label"
  >
    <span
      class="select-label-text"
    >
      SORT BY:
    </span>
    <select
      class="select-component"
    >
      <option
        class="select-option"
        value="date"
      >
        RELEASE DATE
      </option>
      <option
        class="select-option"
        value="title"
      >
        TITLE
      </option>
    </select>
  </label>
</div>
`;
